src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -41,9 +42,39 @@
     if (CollectionUtils.isEmpty(symbols)) {
       return Collections.emptyList();
     } else {
-      return iexClient.getLastTradedPriceForSymbols(symbols.toArray(new String[0]));
+      return iexClient.getLastTradedPriceForSymbols(symbols.toArray(new String[0]),
+          System.getenv("API_TOKEN"));
     }
   }
 
+  /**
+   * Get historical price data (close, high, low, open, and volume) for the given symbol
+   * over a specified time range. See https://iexcloud.io/docs/api/#historical-prices.
+   * @param symbol A string representing a stock symbol for which to retrieve historical data.
+   * @param range A string specifying a range of time. See link.
+   * @param date A string representing a date in YYYYMMDD format. See link.
+   * @return A list of IexHistoricalPrice objects for the symbol for each date in the range of time.
+   */
+
+  public List<IexHistoricalPrice> getHistoricalPrices(
+      final String symbol,
+      final String range,
+      final String date) {
+
+    if (symbol.isEmpty()) {
+      return Collections.emptyList();
+    }
+
+    StringBuilder rangeAndDateBuilder = new StringBuilder(range);
+    if (!(date.isEmpty())) {
+      rangeAndDateBuilder.append("/").append(date);
+    }
+    String rangeAndDate = rangeAndDateBuilder.toString();
```

Comment:
We shouldn't need to do this.... if we have range as a path parameter then we don't need this string building

Reply:
See above